src/dataflow/src/source/postgres.rs

```diff
@@ -141,9 +142,9 @@ impl PostgresSourceReader {
                     .iter()
                     .map(|t| (t.relation_id, t.clone())),
             ),
-            connector,
+            connector: connector.clone(),
             lsn: 0.into(),
-            metrics: PgSourceMetrics::new(metrics, source_id),
+            metrics: PgSourceMetrics::new(metrics, source_id, &connector.details),
```

Comment:
nit: technically this is an expensive clone, can you create the `PgSourceMetrics` in a variable before to avoid it?